### PR TITLE
PROJECT-003:User Auth working

### PR DESCRIPTION
**Pull request by BrewerDeveloping**
## User Auth

- User Auth is now working I have many other files also added correlating with my weather app, I am debating rolling my brewingweather_app (Django) into the user_app(Django). The current structure seems complex to get the URLs and views working together but I want to give it another go tomorrow before rolling them together as a lot of docs I read suggest separating a user auth app from your actual working app.

### CODE REVIEW

File: brewingweather/backend/brewingweather/settings.py [hunk 1]
```diff
@@ -20,12 +24,12 @@
...
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'django-insecure-_dw$fb&*%fb*v$$mkb1j#$%_rsi6n9^257*rx1k12d2%h+pr*@'
+SECRET_KEY = os.environ['SECRET_KEY']
 
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['localhost', '127.0.0.1']
```
Comment (fravila08) [inline brewingweather/backend/brewingweather/settings.py]: I wouldn't mess with alloed hosts just yet but if it works it works

File: brewingweather/backend/brewingweather/urls.py
Comment (fravila08) [inline brewingweather/backend/brewingweather/urls.py:1]: Great url naming conventions

File: brewingweather/backend/brewingweather_app/models.py
Comment (fravila08) [inline brewingweather/backend/brewingweather_app/models.py:1]: may be a good idea to add validators and unique or blank constraints specially for the user model

File: brewingweather/frontend/brewingweather/src/App.jsx [hunk 2]
```diff
@@ -1,35 +1,39 @@
...
+export const UserContext = createContext(null);
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [user, setUser] = useState(null);
+
+  getToken();
+
+  useEffect(() => {
+    const getCurrUser = async () => {
```
Comment (fravila08) [inline brewingweather/frontend/brewingweather/src/App.jsx:18]: Maybe getCurrUser could be exported and utilized as a loader function for App.jsx and then passed through the program through either prop drilling (not ideal) or useContext

**Review by fravila08 — APPROVED** [review]
Really good request overall and I like your React practices. Just like you created utilities for your components it lmay be a good call to add utilities to your Django Apps to interact with your views

File: brewingweather/backend/brewingweather/settings.py (hunk 1 above)
Comment (BrewerDeveloping) [reply brewingweather/backend/brewingweather/settings.py]: I will revert back to blank allowed hosts, it does work and was referenced multiple times in some documentation for while in development

**Pull request merged**